src/components/TimePicker/TimePicker.st.css

```diff
@@ -0,0 +1,90 @@
+:import {
+  -st-from: "wix-ui-core/dist/src/components/TimePicker/TimePicker.st.css";
+  -st-default: TimePicker;
+}
+
+:import {
+  -st-from: "../../palette.st.css";
+  -st-named:
+    disabled,
+    buttonSecondaryText,
+    inputBorderColor, inputBackground,
+    inputHover, inputDisabledColor;
+}
+
+:import {
+  -st-from: "../Input/Input.st.css";
+  -st-default: Input;
+}
+
+:import {
+  -st-from: "../Input/InputStates.st.css";
+  -st-named: input-large, input-medium, input-small;
+}
+
+.root {
+  -st-extends: TimePicker;
+  -st-mixin: Input;
+  -st-states: disabled, size(enum(large, medium, small)),
+    inputWidth(enum(small, medium, large, small_ampm, medium_ampm, large_ampm));
+
+  display: inline-flex;
+  width: auto;
+}
+
+.root:size(large) {
+  -st-mixin: input-large;
+}
+
+.root:size(medium) {
+  -st-mixin: input-medium;
+}
+
+.root:size(small) {
+  -st-mixin: input-small;
+}
+
+.root:inputWidth(small)::nativeInput {
+    width: 47px;
```

Comment:
All of the Input components we have are stretching to 100% according to specs but this one isn't for some reason.
worth checking it out with the UX guys

Reply:
@milkyfruit will tell, but in current design spec there are specific width variations
https://app.zeplin.io/project/5864e02695b5754a69f56150/screen/587b4160691df22f04ff138f

and im not really sure that all input component should stretch 100% by default, it will require consumers to either set `width` prop or do additional wrapper

Reply:
Look at the "Constraints" part of the design spec (bottom left), where it says "3-Columns" you can see that the space between the time and the pickers is marked as "flexible". That to me sounds like 100% width.

Reply:
so for that case there's width prop. Im pretty sure most consumers don't need to deal with timepicker width and so it has sane defaults. In case they want to deal with it, there's width prop, which can also be `100%`

Reply:
It's weird to me to control the style of a component through named props. And why only width?

In any case, like Shlomi said, all other input components have width of 100% so for consistency's sake this should too (or change all others to some other consistent behavior).

Reply:
The component is stretchable:
![image](https://user-images.githubusercontent.com/1684587/42631228-34e679b4-85e2-11e8-9a99-cff39a660f6d.png)

Though, it has a min-width as well:
![image](https://user-images.githubusercontent.com/1684587/42631242-43b57eb8-85e2-11e8-9d6f-2a362524649e.png)
Which variates based on used properties (size, AM/PM, ticker)

It would be handy to allow pick both options: width=min-width and width=100%.
Probably, ideal would be to set a default as stretchable and have a toggle `set width to minimum`.

Reply:
That's fine by me

Reply:
possible options from this discussion:

1. `<TimePicker/>` takes all available space and doesn't control width
on it's own. there is no `width` prop and consumer controls width by
wrapping component and styling that wrapper accordingly

2. `<TimePicker/>` takes only the space that is necessary to display
   value, it adjusts on its own depending on whether am/pm is shown or
   not, depending on `size` chosen. if consumer wants, they can adjust
   width through `width` prop. they can do `width="100%"` to make it flexible if they so choose.

option 2 is currently in this PR.

@shlomitc which one do you choose?